Transcribe of cool1d_multi_g

ChristopherBignamini opened the pull request:
This branch has been created from @mabruzzo gen2024-transcribe branch (https://github.com/mabruzzo/grackle/tree/gen2024-transcribe)

mabruzzo commented on src/clib/cool1d_multi_g-cpp.h:
```diff
@@ -0,0 +1,33 @@
+  grackle::impl::LogTLinInterpScratchBuf logTlininterp_buf,
+  grackle::impl::Cool1DMultiScratchBuf cool1dmulti_buf,
+  grackle::impl::CoolHeatScratchBuf coolingheating_buf
+);
```
I have been following the convention, where I put all functions (without C linkage) inside of the ``grackle::impl`` namespace

```suggestion

namespace grackle::impl {

void cool1d_multi_g(
  int imetal, int iter, //double* comp1, double* comp2, 
  double* edot,
  double* tgas, double* mmw, double* p2d, double* tdust, double* metallicity,
  double* dust2gas, double* rhoH, gr_mask_type* itmask,
  gr_mask_type* itmask_metal, chemistry_data* my_chemistry,
  chemistry_data_storage* my_rates, grackle_field_data* my_fields,
  photo_rate_storage my_uvb_rates, InternalGrUnits internalu,
  IndexRange idx_range,
  grackle::impl::GrainSpeciesCollection grain_temperatures,
  grackle::impl::LogTLinInterpScratchBuf logTlininterp_buf,
  grackle::impl::Cool1DMultiScratchBuf cool1dmulti_buf,
  grackle::impl::CoolHeatScratchBuf coolingheating_buf
);

} // namespace grackle::impl
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.h:
```diff
@@ -0,0 +1,33 @@
+// cool1d_multi_g function from FORTRAN to C++
+
+#ifndef MY_FILE_CPP_H
+#define MY_FILE_CPP_H
```
```suggestion
#ifndef COOL1D_MULTI_G_HPP
#define COOL1D_MULTI_G_HPP
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.h:
```diff
@@ -0,0 +1,33 @@
+  grackle::impl::CoolHeatScratchBuf coolingheating_buf
+);
+
+#endif /* MY_FILE_CPP_H */
```
```suggestion
#endif /* COOL_1D_MULTI_G_HPP */
```

mabruzzo commented on src/clib/Make.config.objects:
```diff
@@ -23,7 +23,7 @@ OBJS_CONFIG_LIB = \
         cool1d_cloudy_g.lo \
         cool1d_cloudy_old_tables_g.lo \
         cool1d_multi_g.lo \
```
This change is required to avoid breaking the classic build-system

```suggestion
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+#include "fortran_func_decls.h"
+#include "utils-cpp.hpp"
+
+void cool1d_multi_g(
```
Related to the earlier namespace suggestion I made:

```suggestion
void grackle::impl::cool1d_multi_g(
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.h:
```diff
@@ -0,0 +1,33 @@
+// See LICENSE file for license and copyright information
+
+/// @file cool1d_multi_g-cpp.h
```
You should rename this file `cool1d_multi_g-cpp.hpp` or `cool1d_multi_g.hpp` (the latter will be acceptable once you remove `cool1d_multi_g.F`

I have been following the convention that I only give files the `.h` suffix when the header can be included by C or C++. If the header can only be included by C++, I use the `.hpp` suffix

(eventually every internal header file will have a `.hpp` suffix)

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  const double mh_local_var = mh_grflt;
+  // approx. mean molecular weight of metals
+  const double mu_metal = 16.;
+  const int ti_max = 20;
```
I think you can delete this line (but we could defer this to a future PR)

```suggestion
  const int ti_max = 20;
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  if (my_chemistry->primordial_chemistry > 0)  {
+
+    if (my_chemistry->self_shielding_method == 0)  // no shielding
+    {
```
Could be deferred to a future PR

```suggestion
    if (my_chemistry->self_shielding_method == 0) { // no shielding
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+            eprintf("NaN in edot[2]:  %d %d %d %g %g %g %g %g %g %g %g %g %g %g %g\n",
+                    i,
+                    idx_range.jp1,
+                    idx_range.kp1,
```
It probably makes more sense for this logging statement to now report 0-based indices now that this code is written in C++.

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  std::vector<double> h2lte(my_fields->grid_dimension[0]);
+  std::vector<double> galdl(my_fields->grid_dimension[0]);
+  std::vector<double> gasgr(my_fields->grid_dimension[0]);
+  std::vector<double> myisrf(my_fields->grid_dimension[0]);
```
```suggestion
  // gas/grain heat transfer rate
  std::vector<double> gasgr(my_fields->grid_dimension[0]);
  // holds values of the interstellar radiation field
  std::vector<double> myisrf(my_fields->grid_dimension[0]);
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  // -----------------------------------------------------------------------
+
+
+  // Arguments
```
```suggestion
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  std::vector<double> alFe3O4(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alAC(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alSiO2D(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alMgO(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alFeS(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alAl2O3(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alreforg(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alvolorg(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> alH2Oice(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> altot(my_rates->gr_N[2-1] * my_fields->grid_dimension[0]);
+  std::vector<double> kpSiM(my_fields->grid_dimension[0]);
+  std::vector<double> kpFeM(my_fields->grid_dimension[0]);
+  std::vector<double> kpMg2SiO4(my_fields->grid_dimension[0]);
+  std::vector<double> kpMgSiO3(my_fields->grid_dimension[0]);
+  std::vector<double> kpFe3O4(my_fields->grid_dimension[0]);
+  std::vector<double> kpAC(my_fields->grid_dimension[0]);
+  std::vector<double> kpSiO2D(my_fields->grid_dimension[0]);
+  std::vector<double> kpMgO(my_fields->grid_dimension[0]);
+  std::vector<double> kpFeS(my_fields->grid_dimension[0]);
+  std::vector<double> kpAl2O3(my_fields->grid_dimension[0]);
+  std::vector<double> kpreforg(my_fields->grid_dimension[0]);
+  std::vector<double> kpvolorg(my_fields->grid_dimension[0]);
+  std::vector<double> kpH2Oice(my_fields->grid_dimension[0]);
+  std::vector<double> kptot(my_fields->grid_dimension[0]);
+
+  // grain temperature
+
+  std::vector<double> gasSiM(my_fields->grid_dimension[0]);
+  std::vector<double> gasFeM(my_fields->grid_dimension[0]);
+  std::vector<double> gasMg2SiO4(my_fields->grid_dimension[0]);
+  std::vector<double> gasMgSiO3(my_fields->grid_dimension[0]);
+  std::vector<double> gasFe3O4(my_fields->grid_dimension[0]);
+  std::vector<double> gasAC(my_fields->grid_dimension[0]);
+  std::vector<double> gasSiO2D(my_fields->grid_dimension[0]);
+  std::vector<double> gasMgO(my_fields->grid_dimension[0]);
+  std::vector<double> gasFeS(my_fields->grid_dimension[0]);
+  std::vector<double> gasAl2O3(my_fields->grid_dimension[0]);
+  std::vector<double> gasreforg(my_fields->grid_dimension[0]);
+  std::vector<double> gasvolorg(my_fields->grid_dimension[0]);
+  std::vector<double> gasH2Oice(my_fields->grid_dimension[0]);
```
## Cleanup of dust-grain species buffer variables (1/3)

#### BACKGROUND

We *need* to "cleanup" these ~55 dust-grain species variables (by aggregating them into a smaller set of variables) before we can transcribe `calc_all_tdust_gasgr_1d_g` and `calc_grain_size_increment_1d`.[^1]
- Originally, I was going to make a bunch of suggestions on how you should aggregate the variables (reusing existing data-structures for tracking similar sets of variables), and let you handle the details on your own. As I dug into how these variables are used (so I could suggest short comments for documenting the physical significance of the variables), I realized that things were messier than I originally thought.
- In particular the names of the `alSiM`, `alFeM`, ..., `alH2Oice` variables are very misleading, and I wanted to fully document them. Because things were so messy, it would also be difficult to communicate exactly what I was thinking here.
- This ultimately led me to write PR #40 so that I could more clearly communicate how I think this should be handled.

There are 3 steps to "cleaning up the dust-grain species variables." Each step is described in a separate comment (this comment describes step 1 of 3)

> [!important]
> I realize that these 3 comments describe a lot of changes. If you prefer, I can make the changes described in these 3 comments, myself in a followup PR

### CHANGES IN THIS COMMENT (STEP 1 of 3)

Once you merge in the most recent changes from the `gen2024-transcribe` branch, can you do the following 4 things (afterwards, it would make sense to check that the code compiles and passes the tests):
1. add `#include "dust_props.hpp"` and `#include "internal_types.hpp"` at the top of this file (if you haven't done so already)
2. can you add the following 4 lines at the very end of this function to properly deallocate memory
  ```c++
    grackle::impl::drop_GrainSpeciesCollection(&grain_temperatures);
    grackle::impl::drop_InternalDustPropBuf(&internal_dust_prop_buf);
    grackle::impl::drop_GrainSpeciesCollection(&grain_kappa);
    grackle::impl::drop_GrainSpeciesCollection(&gas_grainsp_heatrate);
  ```
3. You will need to preform a bunch of substitutions. (Many of the locations where you preform the substitutions will be overwritten when you start using the wrapper functions, but this is the most sensible way to sequence these commits).
   - To make things as easy as possible, you can perform all of the substitutions with the following `sed` command:
     ```sh
     ~/grackle/src/clib $ sed -f cleanup.sed -i \.backup cool1d_multi_g-cpp.C
     ```
   - The contents of the `cleanup.sed` file can be downloaded from [here](https://pastebin.com/QjNULH1Q)
4. Can you replace the ~60 highlighted lines with:

```suggestion
  // buffers of intermediate quantities used within dust-routines (for
  // calculating quantites related to heating/cooling)
  grackle::impl::InternalDustPropBuf internal_dust_prop_buf =
      grackle::impl::new_InternalDustPropBuf(
        my_fields->grid_dimension[0], my_rates->gr_N[1]
      );

  // opacity coefficients for each dust grain (the product of opacity
  // coefficient & gas mass density is the linear absortpion coefficient)
  grackle::impl::GrainSpeciesCollection grain_kappa =
    grackle::impl::new_GrainSpeciesCollection(my_fields->grid_dimension[0]);

  // closely related to grain_kappa
  std::vector<double> kappa_tot(my_fields->grid_dimension[0]);

  // holds the gas/grain-species heat transfer rates
  grackle::impl::GrainSpeciesCollection gas_grainsp_heatrate =
    grackle::impl::new_GrainSpeciesCollection(my_fields->grid_dimension[0]);
```

[^1]: Technically, we don't **need** to aggregate the variables before we transcribe the highlighted functions. We **could** transcribe the highlighted functions first and then aggregate them later. But, the variables definitely need to be aggregated. 
    - By aggregating the variables before transcription, the transcription scripts can propagate the aggregation choices for us into the transcribed function.
    - If we transcribe first and aggregate later, we would need to manually aggregate the variables here, in the transcribed functions, and in every function that the variables are passed along to (clearly this is involves substantially more work)

ChristopherBignamini replied:
As discussed during today's call, grackle::impl::drop_GrainSpeciesCollection(&grain_temperatures); is not needed because memory management for grain_temperatures is done outside the cool1d_multi_g function.

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+             alSiM.data(), alFeM.data(), alMg2SiO4.data(), alMgSiO3.data(), alFe3O4.data(), alAC.data(),
+             alSiO2D.data(), alMgO.data(), alFeS.data(), alAl2O3.data(),
+             alreforg.data(), alvolorg.data(), alH2Oice.data(), altot.data()
+          );
```
## Cleanup of dust-grain species buffer variables (2/3)

After you complete step 1 of the "dust-grain species buffer variable cleanup", can you replace this with a call to `grackle::impl::fortran_wrapper::calc_grain_size_increment_1d`? I can take an educated guess as to what the argument list should look like if that helps

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+                 gasMgSiO3.data(), gasFe3O4.data(), gasAC.data(), gasSiO2D.data(), gasMgO.data(),
+                 gasFeS.data(), gasAl2O3.data(), gasreforg.data(), gasvolorg.data(),
+                 gasH2Oice.data()
+               );
```
## Cleanup of dust-grain species buffer variables (3/3)

After you complete step 2 of the "dust-grain species buffer variable cleanup", can you replace this with a call to `grackle::impl::fortran_wrapper::calc_all_tdust_gasgr_1d_g`? I can take an educated guess as to what the argument list should look like if that helps

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+
+  // Add dust opacity.
+  // if (idspecies .eq. 0), dust opacity is overestimated at Td > 50 K
+  // We better not include dust opacity.
```
```suggestion
  // Add contributions from dust opacity to alpha, the linear absorption coefficient
  //
  //  The original Fortran version of this function had the following 2 comments:
  //    ! if (idspecies .eq. 0), dust opacity is overestimated at Td > 50 K
  //    ! We better not include dust opacity.
  // It's a little unclear how relevant these comments actually are.
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  if ((anydust != MASK_FALSE) && (my_chemistry->dust_species > 0))  {
+    for (i = idx_range.i_start + 1; i<=(idx_range.i_end + 1); i++) {
+      if ( itmask_metal[i-1] != MASK_FALSE )  {
+
```
```suggestion

        // In the future, we should consider renaming `alphad`. The
        // current name is a little confusing since:
        // - the related `alpha` variable holds linear absorption
        //   coefficients (which is commonly denoted by the Greek
        //   letter alpha)
        // - in contrast, `alphad` only ever holds the sum of
        //   opacity coefficients (commonly denoted by the Greek
        //   letter kappa)

```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  tbase1   = internalu.tbase1;
+  xbase1   = internalu.uxyz/(internalu.a_value*internalu.a_units);    // uxyz is [x]*a      = [x]*[a]*a'        '
+  dbase1   = internalu.urho*std::pow((internalu.a_value*internalu.a_units),3); // urho is [dens]/a^3 = [dens]/([a]*a')^3 '
+  coolunit = (std::pow(internalu.a_units,5) * std::pow(xbase1,2) * std::pow(mh_local_var,2)) / (std::pow(tbase1,3) * dbase1);
```
This particular logic had been duplicated between 4 or 5 different Fortran files. I designed the `internalu` related functionality to factor out this exact functionality (I took care to preserve the order of operations to prevent drift in answer-test results). I think it is probably worth doing this before the logic is able to start diverging.

The following suggestion shows how to integrate the `internalu` logic (if you want to do this). If you choose to make this change, you should remove the declaration of `xbase1` and `dbase1` variables, and explicitly confirm that this doesn't break any tests. 

```suggestion
  dom      = internalu_calc_dom_(internalu);
  dom_inv  = 1./dom;
  tbase1   = internalu.tbase1;
  coolunit = internalu.coolunit;
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  int iden, item, itab;
+  std::vector<double> cieY06(my_fields->grid_dimension[0]);
+
+  // opacity table
```
I'm pretty sure this comment was describing argument declarations that are no longer present

```suggestion
```

mabruzzo commented on src/clib/cool1d_multi_g-cpp.C:
```diff
@@ -0,0 +1,1740 @@
+  // Slice locals
+ 
+
+  // Cooling/heating slice locals
```
This comment is stating the obvious, but it's probably worth including (so that anybody who looks at this code understands the long term goal)

```suggestion
// Performing heap allocations for all of the subsequent buffers within this
  // function is a major impediment to (i) CPU performance and (ii) adding GPU
  // support to Grackle. Future work should work on addressing this
  // - in the immediate short-term, we need to focus on aggregating these
  //   variables into logically organized structs. In a lot of cases, it
  //   may make sense to move the buffers into the existing
  //   Cool1DMultiScratchBuf or CoolHeatScratchBuf structs.
  // - in the longer term the goal is to refactor this logic to remove as many
  //   of these buffers as possible (without crippling cache performance on
  //   CPUs)
```

mabruzzo reviewed (changes requested):
I meant to review this a little sooner -- sorry!

**This looks fantastic! Thank you so much for doing this!**

There are handful of things that need to be addressed before this can be merged into either `gen2024` or `gen2024-transcribe`. The vast majority of these things are stuff things that I didn't tell you about ahead of time (and they have to do with some basic conventions I have started following, or lessons that I've learned while transcribing) 

I left a handful of comments requesting changes (a bunch of them are things that I didn't really document, but are conventions that I have been following).

There are 3 other larger-scale changes that would be useful:
1. could you please remove `cool1d_multi_g.F` from the repository?
2. Would you mind creating a commit that remove as many references relating to addition/subtraction an offset of 1 and the `j` and `k` indices?
    - I think we actually talked about doing this. I was probably much too vague
    - For context: the `j` and `k` variables are tracked by `idx_range`. As you may recall, the struct members `idx_range.jp1` and `idx_range.kp1` are abbreviations for "j plus 1" and "k plus 1". This was all done to reduce zero-indexed variables (and we don't think about adding/subtracting one)
    - this can base  that just consists of performing search-and-replace that performs the substitutions `idx_range.jp1-1` -> `idx_range.j` and `idx_range.kp1-1` -> `idx_range.k`.
    - It will be a lot easier to do now, than if we wait until later since the formatting of the strings of text that need to be replaced are currently very consistent (they were auto-generated).
    - If transcription worked properly, there should no longer be any references to `idx_range.kp1` or `idx_range.jp1` (if any references persist, that may be a sign that something went wrong during transcription. Alternatively this could just happen in places where you modified the code)
3. Remove all occurrences of `i-1` from from `cool1d_multi_g-cpp.C`. This is going to involve modifying the for-loops so that `i` is a 0-based index.
	- this can all be accomplished via search-and-replace operations.
	- In case you find it useful, I wrote a simple command-line tool to help do this (it requires a little manual intervention). You can find it [here](https://github.com/mabruzzo/grackle_transcribe/blob/main/tools/index_replacer.py). I think the `--help` command does a reasonable job explaining how to use the tool (but, if you're interested I can provide more elaboration)
4. **After you merge in** the latest changes from `gen2024-transcribe`, it would be great if you could replacing invocations of Fortran routines with wrapper functions. There are some minor differences in the wrapper function's interface (to reflect the planned C++ function's interface), but most should be fairly to straightforward to understand. The easiest functions to replace are:
    - `grackle::impl::fortran_wrapper::calc_temp1d_cloudy_g` easily replaces `FORTRAN_NAME(calc_temp1d_cloudy_g)`
    - `grackle::impl::fortran_wrapper::interpolate_2d_g` easily replaces  `FORTRAN_NAME(interpolate_2d_g)`
    - `grackle::impl::fortran_wrapper::interpolate_3d_g` easily replaces each occurrence of `FORTRAN_NAME(interpolate_3d_g)`
    - as an aside: whenever I have a function that calls multiple wrapper functions, I commonly add namespace `f_wrap = ::grackle::impl::fortran_wrapper;` near the start of the function (i.e. the wrapper-namespace is shortened to `f_wrap` within the function's scope). But obviously, this isn't necessary (you should do whatever you find most useful)

   I left comments that describe the order in which you should replace `FORTRAN_NAME(calc_grain_size_increment_1d)` and `FORTRAN_NAME(calc_all_tdust_gasgr_1d_g)`.[^1] I think that this should cover all fortran functions with C++ wrappers.[^2]

Again, this looks great! I really appreciate the help! This goes without saying, but If you disagree with any of my suggestions, you should obviously feel free to push back!

[^1]: these 2 routines are little more tricky to replace since they require the use of a datatype that doesn't exist yet in your current version of `cool1d_multi_g`
[^2]: If you think it would be useful, you should obviously feel free to add more wrapper functions.

mabruzzo committed:
Updating python docs.

mabruzzo committed:
This patch tries switching the theme to 'furo'

This is an extremely popular theme. While I really like how it looks, that's not my main motivation.

My motivation is formatting of wide-tables
- this is a problem with the ReadTheDocs theme.
  - The RTD theme made a deliberate design decision to make all wide tables scrollable (rather than supporting wraparound) without a straight-forward way to disable it.
  - This is a long-standing problem. See this 10 year old issue, readthedocs/sphinx_rtd_theme#117.
  - While this issue suggests some workarounds (we actually use one of them in Enzo-E), there are some flaws with them.[^1]
- In comparison, the furo theme handles wide tables much more elegantly.

To compare them, see the table at the top of the page titled "Integrating Grackle into your Application"

If we dislike Furo, we can just stick with ReadTheDocs. I can try my hand at implementing the fix again. (I just thought I would suggest this patch before adopting a slightly hacky solution)

[^1]: I actually tried adding one of the fixes to Grackle like 9 months ago, and there was an issue. I'm fuzzy on the details. I think it had to do with sizing of the table on different devices (and it was inconsistent between local builds and a ReadTheDocs build).

mabruzzo committed:
Move test-data retrieval logic into ``scripts/ci/fetch-test-data.py``

The purpose is to let us reuse this logic across multiple contexts:
1. ordinary continuous integration (implemented in this commit)
2. to have a simple way to describe to people how they should get the
   test-data (implemented in this commit)
3. To run tests on binary wheels built in continuous integration (I
   haven't done this yet)

I'm not thrilled that an ~80 line script is replacing ~5 shell commands,
but it makes some sense given the following functionality that I added:
- basic error-handling
- logic to fall back to `curl` when `wget` isn't installed
- conditional suppression of progress bars. Given the size of the
  downloaded file, it's useful to show a progress bar (but if we're in
  CI we suppress it)
- checksum validation

mabruzzo committed:
Use builtin GitHub feature to nicely redirect to Code of Conduct

By adding this file, GitHub adds a nicely labeled tab when you scroll
down to the README

Numpy and Scipy both do something similar (but I'm honestly not sure
where the file that does this is in the Numpy repository)

mabruzzo committed:
lay foundations for cibuildwheel

mabruzzo committed:
introduce some github actions for creating/publishing wheels

mabruzzo committed:
lay groundwork for uploading to test-pypi

mabruzzo committed:
tweak wheels workflow

mabruzzo committed:
some extraneous tweaks

mabruzzo committed:
finish commenting out upload job

mabruzzo committed:
configure workflow to run while we are drafting this PR

mabruzzo committed:
resolving some bugs in the GitHub action

mabruzzo committed:
try to address the error with building wheels on macOS 14

mabruzzo committed:
I think this will fix wheels for MacOS on ARM (for real this time)

mabruzzo committed:
fix issues with uploading wheels

mabruzzo committed:
support builds on macOS-13

mabruzzo committed:
Run subset of pygrackle-test in cibuildwheel & with sdist

(We can enable this now that we rebased this branch to include the
changes from PR #335)

mabruzzo committed:
update exclusion list for sdist

mabruzzo committed:
fix the test_get_grackle_version test for checking sdist installations

mabruzzo committed:
stop using subprocess in test_models

brittonsmith committed:
Merge pull request #329 from mabruzzo/numpy-headers

remove a line that was made obsolete by PR #224

brittonsmith committed:
Merge pull request #319 from mabruzzo/minor-pyproject-tweak

cython does not need to be a runtime-dependency

brittonsmith committed:
Merge pull request #309 from mabruzzo/pygrackle-test-instruction-tweaks

Pygrackle test instruction tweaks

brittonsmith committed:
Merge pull request #326 from mabruzzo/fix-cmake-typo

fix a typo in a CMake variable name

brittonsmith committed:
Merge pull request #328 from mabruzzo/cython-cmake

Tweak  cython-handling in build-system

mabruzzo committed:
Merge branch 'main' into Enable-Compiler-Warnings

mabruzzo committed:
remove unused variables

mabruzzo committed:
a minor bugfix

mabruzzo committed:
Merge branch 'main' into doc-tweaks

brittonsmith committed:
Merge pull request #334 from mabruzzo/Enable-Compiler-Warnings

Enable Compiler Warnings in a CI job

brittonsmith committed:
Merge pull request #325 from mcsmithastro/H2_self_shielding_length_fix

H2 self shielding length scale factor fix

brittonsmith committed:
Fix spacing issues in docs.

brittonsmith committed:
Merge pull request #285 from mabruzzo/doc-tweaks

Superficial Website Documentation Tweaks

brittonsmith committed:
Merge pull request #324 from brittonsmith/model-comp2

[newchem-cpp] Minor model test refactor and adding model comparison option

brittonsmith committed:
Merge branch 'newchem-cpp' of github.com:grackle-project/grackle into ncc-answers

brittonsmith committed:
Merge branch 'main' into newchem-cpp

brittonsmith committed:
Continuing merge with main.

mabruzzo committed:
fixing up config.yml

mabruzzo committed:
removed unused variables

brittonsmith committed:
Merge pull request #45 from mabruzzo/ncc-main-merge-tweaks

fixing up config.yml

mabruzzo committed:
address the remaining compiler warnings

brittonsmith committed:
Merge pull request #46 from mabruzzo/ncc-main-merge-tweaks

address the remaining compiler warnings

mabruzzo committed:
Merge pull request #340 from brittonsmith/spaces

Fix spacing issues in docs.

brittonsmith committed:
Move import to the right function.

mabruzzo committed:
Merge branch 'main' into dependencies-cmake

brittonsmith committed:
Merge pull request #296 from mabruzzo/dependencies-cmake

Relocated all CMake-dependency-information into dependencies.cmake

mabruzzo committed:
ncc-main-merge-bugfix

brittonsmith committed:
Merge pull request #341 from brittonsmith/ncc-main-merge

[newchem-cpp] Attempted merge of newchem-cpp with main

brittonsmith committed:
Merge branch 'newchem-cpp' into ncc-answers

brittonsmith committed:
Merge pull request #342 from mabruzzo/code-of-conduct-redirect

Use builtin GitHub feature to nicely redirect to Code of Conduct

brittonsmith committed:
Merge pull request #300 from mabruzzo/source-links

Simplify GitHub hyperlinks for repository locations in docs

brittonsmith committed:
Merge pull request #327 from mabruzzo/chemistry_data_member_test_modify

Move `test_chemistry_struct_synched` between test suites

brittonsmith committed:
Merge branch 'main' into try-furo

mabruzzo committed:
remove run_command and address linting errors.

mabruzzo committed:
silence noisy linter warning in a particular location (it only became an issue after we enclosed the logic within a function)

brittonsmith committed:
Add caching of gold standard answers.

mabruzzo committed:
Merge pull request #291 from mabruzzo/try-furo

This patch tries switching the theme to 'furo'

mabruzzo committed:
Remove spurious indentations (i.e. stop formatting as block-quotes)

This patch goes through and simply removes spurious indentations. (**NO
TEXT HAS BEEN CHANGED**). This fixes rendering of text.

In more detail, the reStructuredText Markup Specification
[notes that](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#toc-entry-26)
all indented blocks of text (without preceding markup) as quote-blocks.
There were many chunks blocks of text that met this definition.

The impetus for this change is our transition to the new "Furo"
sphinx-theme since Furo applied special formatting to quote-blocks (the
older themes did not do this).

mabruzzo committed:
remove spurious indentations from Parameters.rst

mabruzzo committed:
make it possible to specify the path to output directory in test_models.

mabruzzo committed:
test_models: start using importlib

mabruzzo committed:
shift the dev dependencies to the dev dependency-group

mabruzzo committed:
a minor tweak

mabruzzo committed:
update the docs

mabruzzo committed:
bump python version tested on circleci to 3.9

mabruzzo committed:
Merge pull request #346 from mabruzzo/removing-doc-indents

Remove spurious indentations (fix rendering of text)

mabruzzo committed:
addressing issues higlighted by CI

mabruzzo committed:
probing the rtd ci problem

mabruzzo committed:
another attempt at fixing rtd CI

mabruzzo committed:
reintroduce temporary dependencies

mabruzzo committed:
Merge pull request #345 from brittonsmith/testcache

Add caching of gold standard answers for ci

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc-speedup-test_models

mabruzzo committed:
adjust the python version used in circleci

mabruzzo committed:
Merge branch 'ncc-speedup-test_models' into ncc-answers

mabruzzo committed:
Update doc/source/Python.rst

Co-authored-by: Britton Smith <brittonsmith@gmail.com>

mabruzzo committed:
Merge pull request #347 from mabruzzo/dependency-group

Dependency group

mabruzzo committed:
start using test dependency-group

brittonsmith committed:
Merge pull request #47 from mabruzzo/ncc-answers-merge

Merge 'ncc-speedup-test_models' into ncc-answers

brittonsmith committed:
Merge pull request #344 from mabruzzo/ncc-speedup-test_models

[newchem-cpp] Speedup test_models.py by direct importing

brittonsmith committed:
Merge pull request #336 from brittonsmith/ncc-answers

[newchem-cpp] Add model tests for newchem-cpp functionality

brittonsmith committed:
Merging.

mabruzzo committed:
another attemp

mabruzzo committed:
fixing test-issues when CI builds the pre-compiled wheels

mabruzzo committed:
the version of hdf5 distributed in the binary wheel now supports the built-in zlib and szip filters

mabruzzo committed:
stop distributing libaec, and more properly handle zlib

mabruzzo committed:
another attempt at getting zlib to play nicely

mabruzzo committed:
another attempt

mabruzzo committed:
trying another tact

mabruzzo committed:
Add a detailed error-message for when pygrackle installation fails using a sdist

mabruzzo committed:
Fix SPDX identifiers in binary_license_annex

I changed 2 parts:
- HDF5: I added the HDF5 SPDX identifier.
- GCC runtime library: I made a change reflecting a recent numpy commit

mabruzzo committed:
Add check_packaged_dependency_licenses.py

This verifies that all libraries included in the wheel have a license entry
and confirm that we didn't include any unnecessary licenses

This is invoked as part of building the binary wheels

mabruzzo committed:
adding a missing pattern for matching libgcc_s*

mabruzzo committed:
bugfix in check_packaged_dependency_licenses

mabruzzo committed:
fix a typo

mabruzzo committed:
try uploading to testpypi

mabruzzo committed:
bugfix

mabruzzo committed:
take a stab at fixing the workflows

mabruzzo committed:
bump minimum required python version and add a check for the README

mabruzzo committed:
Apply suggestions from code review

Co-authored-by: Britton Smith <brittonsmith@gmail.com>

mabruzzo committed:
improving clarity of the documentation

mabruzzo committed:
minor tweak

brittonsmith committed:
Merge pull request #294 from mabruzzo/pydoc-update

Add more detail to pygrackle build documentation

mabruzzo committed:
configure the ruff linter so that it is consistent with flake8

I also fixed some of the errors it found

mabruzzo committed:
Update the documentation for pre-commit stuff

mabruzzo committed:
add more details about how wheel-packaging works

mabruzzo committed:
Introduce an extremely simplistic .pre-commit-config.yaml file

The intention here is to merge this commit into the main branch while we
configure pre-commit to run code formatting/linting in the newchem-cpp
branch. PR #287 (where we start configuring pre-commit to run
formatting/linting) is configured so all changes build off this commit
(in other words, there won't be any merge conflicts when we ultimately
merge newchem-cpp into main).

If we don't merge a .pre-commit-config.yaml file into the main branch,
then pre-commit.ci will report a failure (since it can't find a
.pre-commit-config.yaml file)

mabruzzo committed:
Introduce an extremely simplistic .pre-commit-config.yaml file

The intention here is to merge this commit into the main branch while we
configure pre-commit to run code formatting/linting in the newchem-cpp
branch. PR #287 (where we start configuring pre-commit to run
formatting/linting) is configured so all changes build off this commit
(in other words, there won't be any merge conflicts when we ultimately
merge newchem-cpp into main).

If we don't merge a .pre-commit-config.yaml file into the main branch,
then pre-commit.ci will report a failure (since it can't find a
.pre-commit-config.yaml file)

mabruzzo committed:
introduce Ruff for linting/formatting python code

mabruzzo committed:
adjust wording in 2 spots

mabruzzo committed:
introduce clang-format for C/C++ code formatting

mabruzzo committed:
fix some undefined behavior and update gold-standard

mabruzzo committed:
tweak the workflow

mabruzzo committed:
Merge branch 'ncc-lookup_cool_rates0d_bugfix' into remove-last-logical

mabruzzo committed:
Merge branch 'ncc-lookup_cool_rates0d_bugfix' into gen2024-calc_grain_size_increment_1d.F

mabruzzo committed:
Merge branch 'remove-last-logical' into gen2024-transcribe_cool_multi_time_g

mabruzzo committed:
Merge branch 'ncc-lookup_cool_rates0d_bugfix' into gen2024-cleanup_calc_tdust_3d_g

mabruzzo committed:
Merge branch 'gen2024-cleanup_calc_tdust_3d_g' into gen2024-refactor_dust_calcs

mabruzzo committed:
Merge branch 'ncc-lookup_cool_rates0d_bugfix' into gen2024-tweak-index_helper_declarations

mabruzzo committed:
Merge branch 'ncc-lookup_cool_rates0d_bugfix' into gen2024-rate_access_API

mabruzzo committed:
Merge branch 'gen2024-transcribe_cool_multi_time_g' into gen2024transcribe/solve_rate_cool_g

brittonsmith committed:
Merge pull request #349 from mabruzzo/ncc-lookup_cool_rates0d_bugfix

[newchem-cpp] Fix undefined behavior in lookup_cool_rates0d and update gold-standard

mabruzzo committed:
Merge branch 'gen2024transcribe/solve_rate_cool_g' into gen2024transcribe/consistency-tweaks

brittonsmith committed:
Merge pull request #348 from mabruzzo/dummy-precommit

Introduce an extremely simplistic `.pre-commit-config.yaml` file

mabruzzo committed:
Merge branch 'gen2024transcribe/consistency-tweaks' into gen2024transcribe/internal_types_updates

mabruzzo committed:
Merge branch 'code-formatting' into gen2024transcribe/internal_types_updates

mabruzzo committed:
transition the README from markdown to restructured text

mabruzzo committed:
Deduplicate the contents of the README and the website's landing page

mabruzzo committed:
lay the foundations for using a better form of the visitor pattern

mabruzzo committed:
migrate internal_types to new machinery

brittonsmith committed:
Merge pull request #320 from mabruzzo/binary-wheel

Packaging Pygrackle

brittonsmith committed:
Merge pull request #350 from mabruzzo/README-to-RST

Convert README.md to README.rst (and deduplicate content on the website landing page)

mabruzzo committed:
prepare for hierarichal visitors

mabruzzo committed:
make SpeciesRateSolverScratchBuf use visitor machinery

mabruzzo committed:
start formatting internal_types.C

mabruzzo committed:
fixing a bug in pyproject.toml that arose because PRs #320 and #350 introduced conflicting changes

mabruzzo committed:
fix a minor bug introduced to classic build-system (related to the visitor subdirectory)

brittonsmith committed:
Merge pull request #351 from mabruzzo/pyproject-hotfix

fixing a bug in pyproject.toml that arose because PRs #320 and #350 introduced conflicting changes

mabruzzo committed:
Remove pygrackle/api.py & pygrackle/utilities/api.py

It seems like we can drop these files. After all, they were deprecated
9 years ago (well before the pygrackle 1.0)

mabruzzo committed:
Merge branch 'gen2024transcribe/internal_types_updates' into gen2024transcribe/Implement-SpLUT-with-XMacros

mabruzzo committed:
start applying clang-format to src/clib/LUT.hpp

mabruzzo committed:
Merge branch 'gen2024transcribe/Implement-SpLUT-with-XMacros' into gen2024transcribe/SpeciesLUTFieldAdaptor

mabruzzo committed:
start applying clang-format to clib/utils-field.hpp

mabruzzo committed:
try bumping the dev version of pygrackle so we can test uploading to pygrackle once before I update the wheel-action so we only try uploading when we update versions

brittonsmith committed:
Merge pull request #355 from mabruzzo/binary-wheel

try bumping the dev version of pygrackle so we can test uploading wheels

brittonsmith committed:
Merge pull request #261 from mabruzzo/remove-last-logical

[newchem-cpp] Remove last occurence of `logical` from Fortran source files

brittonsmith committed:
Merge pull request #274 from mabruzzo/gen2024-calc_grain_size_increment_1d.F

[newchem-cpp] tweak `calc_grain_size_increment_1d.F`

brittonsmith committed:
Merge pull request #276 from mabruzzo/gen2024-cleanup_calc_tdust_3d_g

[newchem-cpp] cleanup `calc_tdust_3d_g`

brittonsmith committed:
Merge pull request #315 from mabruzzo/gen2024-tweak-index_helper_declarations

[newchem-cpp] tweak `index_helper.h` for use with transcribed routines (and general C++ compatibility)

brittonsmith committed:
Merge pull request #352 from mabruzzo/remove-api.py

Remove pygrackle/api.py & pygrackle/utilities/api.py

mabruzzo committed:
stop trying to upload to TestPyPI after every PR

mabruzzo committed:
Merge branch 'main' into introduce-status-reporting

brittonsmith committed:
Merge pull request #356 from mabruzzo/fix-action

Modify GHA to stop trying to upload to TestPyPI after every PR

mabruzzo committed:
Rename test-data fetcher script and remove duplicated code

In more detail, I renamed fetch-test-data.py -> fetch_test_data.py, and
at Britton's suggestion, I stopped calling command line programs from
within that script.

While I was here, I removed some duplicated code from
cibw_before_build.py and imported it from fetch_test_data.py

mabruzzo committed:
Merge pull request #277 from mabruzzo/gen2024-refactor_dust_calcs

[newchem-cpp] factor out shared logic from `cool1d_multi_g` and `calc_tdust_3d_g`

mabruzzo committed:
Apply suggestions from code review

Co-authored-by: Britton Smith <brittonsmith@gmail.com>

brittonsmith committed:
Merging with main.

mabruzzo committed:
Merge branch 'main' into precommit-and-ruff

mabruzzo committed:
Apply additional suggestions from code review

Co-authored-by: Britton Smith <brittonsmith@gmail.com>

mabruzzo committed:
Merge branch 'newchem-cpp' into precommit-and-ruff

mabruzzo committed:
Update doc/source/CodeFormatting.rst

Co-authored-by: Britton Smith <brittonsmith@gmail.com>

mabruzzo committed:
Merge branch 'precommit-and-ruff' into code-formatting

mabruzzo committed:
address 1 last fix

mabruzzo committed:
Merge branch 'precommit-and-ruff' into code-formatting

mabruzzo committed:
Merge pull request #287 from mabruzzo/precommit-and-ruff

Introduce pre-commit with ``ruff`` for formatting new python files

mabruzzo committed:
Merge pull request #222 from mabruzzo/code-formatting

C/C++ Code formatting

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024-transcribe_cool_multi_time_g

mabruzzo committed:
Merge pull request #307 from mabruzzo/fetch-test-data.py

Move test-data retrieval logic into ``scripts/ci/fetch-test-data.py``

mabruzzo committed:
Merge branch 'main' into introduce-status-reporting

mabruzzo committed:
shift fflush and fix some typos in docstrings

mabruzzo committed:
shift fflush and fix some typos in docstrings

mabruzzo committed:
Merge pull request #269 from mabruzzo/introduce-status-reporting

Standard Status Reporting Machinery

mabruzzo committed:
Merge pull request #275 from mabruzzo/gen2024-transcribe_cool_multi_time_g

[newchem-cpp] initial transcription of `cool_multi_time_g`

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024transcribe/solve_rate_cool_g

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024-rate_access_API

mabruzzo committed:
change name from ColRecRxnRates to CollisionalRxnRates

mabruzzo committed:
start formatting src/clib/rate_utils.c

mabruzzo committed:
pull in the most recent change from gen2024-rate_access_API

mabruzzo committed:
address an oversight

mabruzzo committed:
Replace ColRecRxn with CollisionalRxn

This is going to make a lot of merge conflicts

mabruzzo committed:
Merge branch 'gen2024transcribe/solve_rate_cool_g' into gen2024transcribe/consistency-tweaks

mabruzzo committed:
Merge branch 'gen2024transcribe/consistency-tweaks' into gen2024transcribe/internal_types_updates

dependabot[bot] committed:
Bump pypa/cibuildwheel from 2.23.3 to 3.0.0 in the actions group

Bumps the actions group with 1 update: [pypa/cibuildwheel](https://github.com/pypa/cibuildwheel).


Updates `pypa/cibuildwheel` from 2.23.3 to 3.0.0
- [Release notes](https://github.com/pypa/cibuildwheel/releases)
- [Changelog](https://github.com/pypa/cibuildwheel/blob/main/docs/changelog.md)
- [Commits](https://github.com/pypa/cibuildwheel/compare/v2.23.3...v3.0.0)

---
updated-dependencies:
- dependency-name: pypa/cibuildwheel
  dependency-version: 3.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: actions
...

Signed-off-by: dependabot[bot] <support@github.com>

mabruzzo committed:
Merge pull request #360 from grackle-project/dependabot/github_actions/actions-d86b61c7bf

Bump pypa/cibuildwheel from 2.23.3 to 3.0.0 in the actions group

brittonsmith committed:
Merge pull request #316 from mabruzzo/gen2024-rate_access_API

[newchem-cpp] rate access api

KeiganM committed:
Added missing He

brittonsmith committed:
Merge pull request #362 from KeiganM/solve

Added missing He

dependabot[bot] committed:
Bump pypa/cibuildwheel from 3.0.0 to 3.0.1 in the actions group

Bumps the actions group with 1 update: [pypa/cibuildwheel](https://github.com/pypa/cibuildwheel).


Updates `pypa/cibuildwheel` from 3.0.0 to 3.0.1
- [Release notes](https://github.com/pypa/cibuildwheel/releases)
- [Changelog](https://github.com/pypa/cibuildwheel/blob/main/docs/changelog.md)
- [Commits](https://github.com/pypa/cibuildwheel/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: pypa/cibuildwheel
  dependency-version: 3.0.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: actions
...

Signed-off-by: dependabot[bot] <support@github.com>

mabruzzo committed:
Merge pull request #363 from grackle-project/dependabot/github_actions/actions-ba9b347b0f

Bump pypa/cibuildwheel from 3.0.0 to 3.0.1 in the actions group

dependabot[bot] committed:
Bump pypa/cibuildwheel from 3.0.1 to 3.1.1 in the actions group

Bumps the actions group with 1 update: [pypa/cibuildwheel](https://github.com/pypa/cibuildwheel).


Updates `pypa/cibuildwheel` from 3.0.1 to 3.1.1
- [Release notes](https://github.com/pypa/cibuildwheel/releases)
- [Changelog](https://github.com/pypa/cibuildwheel/blob/main/docs/changelog.md)
- [Commits](https://github.com/pypa/cibuildwheel/compare/v3.0.1...v3.1.1)

---
updated-dependencies:
- dependency-name: pypa/cibuildwheel
  dependency-version: 3.1.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: actions
...

Signed-off-by: dependabot[bot] <support@github.com>

mabruzzo committed:
Merge pull request #365 from grackle-project/dependabot/github_actions/actions-d6c157a090

Bump pypa/cibuildwheel from 3.0.1 to 3.1.1 in the actions group

mabruzzo committed:
address a bug calc_temp_cloudy_g.F

mabruzzo committed:
start testing with newer gold-standard

mabruzzo committed:
backport test_get_grackle_version.py changes from newchem-cpp

It's **important** that we introduce this commit **before** we bump the
gold-standard

mabruzzo committed:
Address linter warnings in __init__.py

I slightly tweaked the format of __init__.py so that the F401 linter
warning won't be an issue when we shift to using ruff-format (we start
using ruff-format in newchem-cpp). I had thought this would also work
with the flake8 linter, but I didn't have any luck.

Consequently, I tweaked setup.cfg to have flake8 process **__init__.py**
while ignoring F401, rather than ignoring the rule entirely.

mabruzzo committed:
adjust pygrackle/utilities/testing.py to satisfy flake8

brittonsmith committed:
Merge pull request #367 from mabruzzo/calcTbugfix

address a bug calc_temp_cloudy_g.F

mabruzzo committed:
The *Great Renaming:* gracklepy edition

brittonsmith committed:
Merge pull request #331 from mabruzzo/gen2024transcribe/solve_rate_cool_g

[newchem-cpp] transcribe: `solve_rate_cool_g` to C++

brittonsmith committed:
Merge pull request #368 from mabruzzo/address-linter-__init__

Address linter warnings in __init__.py

dependabot[bot] committed:
Bump pypa/cibuildwheel from 3.1.1 to 3.1.3 in the actions group

Bumps the actions group with 1 update: [pypa/cibuildwheel](https://github.com/pypa/cibuildwheel).


Updates `pypa/cibuildwheel` from 3.1.1 to 3.1.3
- [Release notes](https://github.com/pypa/cibuildwheel/releases)
- [Changelog](https://github.com/pypa/cibuildwheel/blob/main/docs/changelog.md)
- [Commits](https://github.com/pypa/cibuildwheel/compare/v3.1.1...v3.1.3)

---
updated-dependencies:
- dependency-name: pypa/cibuildwheel
  dependency-version: 3.1.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: actions
...

Signed-off-by: dependabot[bot] <support@github.com>

brittonsmith committed:
Merge pull request #372 from grackle-project/dependabot/github_actions/actions-6d8113c782

Bump pypa/cibuildwheel from 3.1.1 to 3.1.3 in the actions group

mabruzzo committed:
minor bugfix

mabruzzo committed:
Merge branch 'main' into address-flake8-other

mabruzzo committed:
address 1 last issue

brittonsmith committed:
Merge pull request #369 from mabruzzo/address-flake8-other

adjust pygrackle/utilities/testing.py to satisfy flake8

mabruzzo committed:
add a note about uninstalling pygrackle before you install gracklepy

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024transcribe/internal_types_updates

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024transcribe/consistency-tweaks

mabruzzo committed:
Merge branch 'gen2024transcribe/internal_types_updates' into gen2024transcribe/Implement-SpLUT-with-XMacros

mabruzzo committed:
Merge branch 'gen2024transcribe/Implement-SpLUT-with-XMacros' into gen2024transcribe/SpeciesLUTFieldAdaptor

brittonsmith committed:
Merge pull request #332 from mabruzzo/gen2024transcribe/consistency-tweaks

[newchem-cpp]  Consistency tweaks: `GrainSpeciesCollection` and `cool_multi_time_g`

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024transcribe/internal_types_updates

mabruzzo committed:
Merge branch 'gen2024transcribe/internal_types_updates' into gen2024transcribe/Implement-SpLUT-with-XMacros

mabruzzo committed:
Merge branch 'gen2024transcribe/Implement-SpLUT-with-XMacros' into gen2024transcribe/SpeciesLUTFieldAdaptor

mabruzzo committed:
Merge branch 'gen2024transcribe/SpeciesLUTFieldAdaptor' into gen2024transcribe/initial-step_rate_newton_raphson

mabruzzo committed:
Merge branch 'gen2024transcribe/initial-step_rate_newton_raphson' into gen2024transcribe/step_rate_newton_raphson-cleanup-1of2

mabruzzo committed:
satisfy clang-format

mabruzzo committed:
tweak Gracklepy docs (to acknowledge historical name)

brittonsmith committed:
Merge pull request #333 from mabruzzo/gen2024transcribe/internal_types_updates

[newchem-cpp] internal scratch-buffer interface update

brittonsmith committed:
Merge pull request #353 from mabruzzo/gen2024transcribe/Implement-SpLUT-with-XMacros

[newchem-cpp] Refactor such that SpLUT is implemented with XMacros

mabruzzo committed:
Merge pull request #354 from mabruzzo/gen2024transcribe/SpeciesLUTFieldAdaptor

[newchem-cpp] Introduce `SpeciesLutFieldAdaptor`

mabruzzo committed:
Merge pull request #371 from mabruzzo/rename-gracklepy

The *Great Renaming:* `gracklepy` edition

brittonsmith committed:
Merge pull request #374 from mabruzzo/gen2024transcribe/initial-step_rate_newton_raphson

[newchem-cpp] initial transcription of `step_rate_newton_raphson`

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024transcribe/step_rate_newton_raphson-cleanup-1of2

mabruzzo committed:
I think this solves the bug

mabruzzo committed:
address compiler warning

mabruzzo committed:
Removed the now-unnecessary check that iter == 1 in newton-raphson

mabruzzo committed:
Merge branch 'main' into newchem-cpp

mabruzzo committed:
bumping ncc gold-standard

mabruzzo committed:
address linter error

mabruzzo committed:
Merge branch 'newchem-cpp' into gen2024transcribe/step_rate_newton_raphson-cleanup-1of2

mabruzzo committed:
Merge branch 'gen2024transcribe/step_rate_newton_raphson-cleanup-1of2' into gen2024transcribe/solve_rate_cool-refactor

mabruzzo committed:
Merge branch 'gen2024transcribe/step_rate_newton_raphson-cleanup-1of2' into gen2024transcribe/lookup_cool_rates0d

mabruzzo committed:
address compiler warning (rm unused vars)

mabruzzo committed:
apply clang-format

mabruzzo committed:
Merge branch 'gen2024transcribe/lookup_cool_rates0d' into gen2024transcribe/lookup_cool_rates0d_cleanup

mabruzzo committed:
Merge branch 'gen2024transcribe/lookup_cool_rates0d_cleanup' into gen2024transcribe/calc_temp_cloudy_g

I needed to introduce the bugfix from PR #367.

mabruzzo committed:
start formatting calc_temp_cloudy_g-cpp.[Ch]

mabruzzo committed:
Merge branch 'gen2024transcribe/calc_temp_cloudy_g' into gen2024transcribe/scale_fields_table

mabruzzo committed:
start formatting scale_fields_table.hpp

mabruzzo committed:
Merge branch 'gen2024transcribe/scale_fields_table' into gen2024transcribe/interpolate-wrappers

mabruzzo committed:
Merge branch 'gen2024transcribe/interpolate-wrappers' into gen2024transcribe/calc_tdust_3d

mabruzzo committed:
Merge branch 'gen2024transcribe/calc_tdust_3d' into gen2024transcribe/scale_fields_dust

mabruzzo committed:
start formatting some of the new files

mabruzzo committed:
Remove last references to Flake8

Background
----------

Previously we had replaced 99% uses of the Flake8 linter with the Ruff
linter (not to be confused with the Ruff formatter). We had continued to
use the Flake8 linter to deal with the following 2 files:

1. **gracklepy/utilities/testing.py**
2. **gracklepy/__init__.py**

In more detail, we were suppressing certain warnings in these 2 files
from Flake8, and transistioning to using the Ruff formatter for these
files would involve making a few changes.

More recently PRs #368 and #369 introduced the required changes into the
main branch. Now that those changes have been ported to the newchem-cpp
branch, we are able to make this change.

What Changed
------------

I essentially changed 3 things:

1. We no longer install and use flake8 in the continuous integration
2. I removed the **setup.cfg** file (it was **only** being used to track
   flake8 configuration information
3. I needed to tweak **pyproject.toml** to both:
   - stop excluding the **gracklepy/utilities/testing.py** and
     **gracklepy/__init__.py** files from the linter
   - stop listing flake8 as a development dependency

brittonsmith committed:
Merge pull request #375 from mabruzzo/gen2024transcribe/step_rate_newton_raphson-cleanup-1of2

[newchem-cpp] `step_rate_newton_raphson` cleanup (1/2)

brittonsmith committed:
Merge pull request #377 from mabruzzo/gen2024transcribe/solve_rate_cool-refactor

[newchem-cpp] light refactoring of `solve_rate_cool`

brittonsmith committed:
Merge pull request #387 from mabruzzo/ncc-remove-flake8

[newchem-cpp] Remove last references to `Flake8`

ChristopherBignamini committed:
Merge remote-tracking branch 'mabruzzo/gen2024transcribe/scale_fields_dust' into gen2024_transcribe_cool1d_multi_g

ChristopherBignamini committed:
Merge remote-tracking branch 'remotes/upstream/newchem-cpp' into gen2024_transcribe_cool1d_multi_g

ChristopherBignamini commented:
Moved to main grackle repo as PR to the newchem_cpp branch:
https://github.com/grackle-project/grackle/pull/405
